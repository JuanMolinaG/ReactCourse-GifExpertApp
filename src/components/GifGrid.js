import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__pulse'>{category}</h3>

      {loading && (
        <p className='animate__animated animate__fadeIn'>Loading...</p>
      )}

      <div className='card-grid'>
        {gifs.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
