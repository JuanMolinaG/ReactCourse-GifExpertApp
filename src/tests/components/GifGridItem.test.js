import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = 'Hola Mundo';
  const url = 'http://localhost/image.jpg';

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe de mostrar el componente <GifGridItem /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un <p> con el title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('Debe de tener un <img/> con la url en el src y el title en el alt', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('El <div> debe de tener la clase animate__pulse', () => {
    const className = 'animate__pulse';
    const div = wrapper.find('div');

    expect(div.hasClass(className)).toBe(true);
  });
});
