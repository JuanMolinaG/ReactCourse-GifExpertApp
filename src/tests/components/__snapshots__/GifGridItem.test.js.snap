// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="Hola Mundo"
    src="http://localhost/image.jpg"
  />
  <p>
    Hola Mundo
  </p>
</div>
`;
